apache-demo/docs/adopt.md

```diff
@@ -0,0 +1,177 @@
+    CLIENT_ID=Admin
+    ```
+    2. Get the $CLIENT_SECRET is the key for clientId: Admin and decode it using base64.
+    ```
+    CLIENT_SECRET=$(oc get secret $TNCO_CREDENTIALS_SECRET -o jsonpath='{.data.adminClientSecret}' -n $NAMESPACE | base64 -d)
+    ```
+    3. Fetch the API PROXY_ADDRESS (Ishtar route):
+    ```
+    PROXY_ADDRESS=$(oc get routes alm-ishtar -o jsonpath='{.spec.host}' -n $NAMESPACE)
+    ```
+    4. Set an encoded header for basic authorization:
+    ```
+    BASIC_AUTH_ENCODED=$(echo -n "$CLIENT_ID:$CLIENT_SECRET" | base64)
+    ```
+    5. Generate the ACCESS_TOKEN that is required for OAuth authentication:
+    ```
+    ACCESS_TOKEN=$(curl -sk -X POST "https://$PROXY_ADDRESS/oauth/token" -H "Authorization: Basic $BASIC_AUTH_ENCODED" -d 'grant_type=client_credentials' | python -c 'import sys, json; print json.load(sys.stdin)["access_token"]')
+    ```
+    You can test the access token by calling the API to list the assemblies, for example:
+    ```
+    curl --insecure -H 'Accept: */*' -X GET -H "Authorization: Bearer $ACCESS_TOKEN" https://$PROXY_ADDRESS:443/api/topology/assemblies
+    ```
+6. Run the Adopt API call to bring the stack under IBM Telco Network Cloud - Orchestration control.  
+    1. Use the following JSON data to adopt the OpenStack stack created in the previous step:
+   
+            {
+                "assemblyName": "apacheadopt",
+                "descriptorName": "assembly::apache-demo::1.0",
+                "properties": {
+                    "resourceManager": "brent",
+                    "deploymentLocation": "test-openstack"
+                }, 
+                "clusters": {
+                },
+                "resources": {
+                    "apacheadopt__hw-apache": {
+                        "properties": {
+                        },
+                        "associatedTopology": [{
+                            "id": "5265adec-503e-4e8a-885a-91ba99ee57b0 ",
+                            "type": "openstack"
+                        }]
+                    }
+                }
+            }
+  
+        **Note:** The resource name under resources conforms to the standard resource instance name syntax, that is, a string of components separated by __ characters. Each component represents a component in the descriptor or a number for clustered components, starting with the assembly instance name. In this example, hw-apache is found in the apache-examples/apache-demo/Descriptor/assembly.yml descriptor.
+    2. Save the JSON content to a file named adoptRequest.json and modify the fields assemblyName, deploymentLocation, resources, and associatedTopology.id as necessary.
+    3. Adopt the stack by using the following REST API call against the API:
+    ```
+    curl -v --insecure -X POST -H "Authorization: Bearer $ACCESS_TOKEN" -H "Content-Type: application/json" https://$PROXY_ADDRESS:443/api/intent/adoptAssembly -d @adoptRequest.json
```

Comment:
If changes in step 5 are implemented - then this command should be updated (using format as per the apache-ansible-demo example) 

```
curl -sk -0 -X POST "https://$TNCO_URL/api/intent/adoptAssembly" -H 'Authorization: Bearer '"${TNCO_TOKEN}"'' -H 'Content-Type: application/json' -d @adoptRequest.json
```